pkg/plugins/plugins.go

```diff
@@ -0,0 +1,137 @@
+package plugins
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/openshift/console/pkg/serverutils"
+	"k8s.io/klog"
+)
+
+type PluginsHandler struct {
+	Client              *http.Client
+	RequestURL          *url.URL
+	ServiceAccountToken string
+	PluginsEndpointMap  map[string]string
+	EnabledPlugins      []string
+}
+
+func NewPluginsHandler(token string, requestURL *url.URL, client *http.Client, enabledPlugins []string) PluginsHandler {
+	p := &PluginsHandler{
+		Client:              client,
+		RequestURL:          requestURL,
+		ServiceAccountToken: token,
+		EnabledPlugins:      enabledPlugins,
+	}
+	return *p
+}
+
+func (p *PluginsHandler) HandlePlugins(w http.ResponseWriter, r *http.Request) {
+
+	if r.Method != "GET" {
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "invalid method: only GET is allowed"})
+		return
+	}
+
+	pluginName, pluginPath := parsePluginNameAndPath(r.URL.Path)
+
+	if !contains(p.EnabledPlugins, pluginName) {
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: fmt.Sprintf("bad request: plugin %q is not enabled", pluginName)})
+		return
+	}
+
+	pluginEndpoint, ok := p.PluginsEndpointMap[pluginName]
+	if ok != true {
+		serverutils.SendResponse(w, http.StatusInternalServerError, serverutils.ApiError{Err: fmt.Sprintf("failed to create GET request: could not get endpoint for %q plugin", pluginName)})
+		return
+	}
+
+	pluginRequest := fmt.Sprintf("https://%s/%s", pluginEndpoint, pluginPath)
```

Comment:
Please also include plugin's "base path" in the constructed URL, which should look like:

```
https://<plugin_endpoint>/<base_path>/<request_path>
```